src/python/pants/base/validation.py

```diff
@@ -23,6 +23,29 @@ def assert_list(obj, expected_type=string_types, can_be_none=True, default=(),
   :param allowable     : the acceptable types for obj. We do not want to allow any iterable (eg string).
   :param raise_type    : the error to throw if the type is not correct.
   """
+  if field_name is not None:
+    val = obj
```

Comment:
Too much copying and pasting.  You can merge it in more elegantly.  Up here, format a temporary variable:

```
if field_name is not None:
  field_msg = "in field'{}'"....
else:
  field_msg = ''
```

Then below, addd field_msg into each of the exception strings:

```
 raise raise_type(
          "{}Expected an object of acceptable type {}, received None and can_be_none is False"
          .format(field_msg, allowable))
  ...
```